Makefile.features

```diff
@@ -22,9 +22,11 @@ endif
 # Resolve FEATURES_ variables
 # Their value will only be complete after resolving dependencies
 
-# Features that are required by the application but not provided by the BSP
+# Features that are required by the application but not provided by the BSP and
+# features that are used but blacklisted (prepended with "!").
 # Having features missing may case the build to fail.
-FEATURES_MISSING = $(sort $(filter-out $(FEATURES_PROVIDED),$(FEATURES_REQUIRED)))
+FEATURES_MISSING = $(sort $(filter-out $(FEATURES_PROVIDED),$(FEATURES_REQUIRED)) \
```

Comment:
You are changing the definition of `FEATURES_MISSING` and not updating the doc:

https://github.com/RIOT-OS/RIOT/blob/3eae2f7b792d60f48006d8338c91a73cb4e0a99a/makefiles/info.inc.mk#L61

Also why is a used blacklisted feature a `FEATURES_MISSING`? To me the usage doesn't fit with the name. This would be more of a  `FEATURES_USED_BLACKLISTED`.

Reply:
Used but blacklisted features get missed as `!<feature>`. Doc is here:

https://github.com/kaspar030/RIOT/blob/3eae2f7b792d60f48006d8338c91a73cb4e0a99a/Makefile.features#L25-L29

There's also `_features_used_blacklisted`.

Reply:
```
Used but blacklisted features get missed as !<feature>. Doc is here:

https://github.com/kaspar030/RIOT/blob/3eae2f7b792d60f48006d8338c91a73cb4e0a99a/Makefile.features#L25-L29
```

I referenced `RIOT/makefiles/info.inc.mk`, not the code.

> ```There's also _features_used_blacklisted.``` 

I'm not arguing this, I'm saying names should make sense with the definiton. A missing `FEATURE` is not a blacklisted used `FEATURE`, why does it make sense adding `_features_used_blacklisted` to `FEATURES_MISSING`. Is this just to re-use the `FEATURES_MISSING` error handling?

If I do `BOARD=esp8266-olimex-mod make -C tests/lwip info-build`, I get:

```
FEATURES_MISSING (only non optional features):
         !arch_esp8266
FEATURES_BLACKLIST (blacklisted features):
         arch_esp8266
```

Before, `FEATURES_MISSING` was self-documented because the name made sense. Now if I do not look at the code, I do not undersand what is going on with `!arch_esp8266`, is `!arch_esp8266` missing? its not missing? (I do not like double negation). In #11492 an effort was made for names to make sense.

Reply:
> Now if I do not look at the code, I do not undersand what is going on with `!arch_esp8266`, is `!arch_esp8266` missing?

Do *you* know or do you assume that is what's gonna happen to someone?
I think if this shows up:

```
FEATURES_MISSING (only non optional features):
         !arch_esp8266
FEATURES_BLACKLIST (blacklisted features):
         arch_esp8266
```

it is pretty clear what's happening.

But yes, this is re-using FEATURES_MISSING.

> A missing FEATURE is not a blacklisted used FEATURE, why does it make sense adding _features_used_blacklisted to FEATURES_MISSING.

Well, it is adding `!feature`, not `feature`, to FEATURES_MISSING. For C devs there should be some meaning there.
Before, features could just be "missing" (e.g., "not provided"). Now, with `!feature`, they can be "used but blacklisted". A more fitting name might be FEATURE_REQUIREMENTS_UNFULFILLED" or something similar.

Again, we're in a situation where the machinery works but the names are not to everyone's liking.
How about we go with the functionality and move the naming struggle elsewhere?

Reply:
> Do you know or do you assume that is what's gonna happen to someone?

Yes since I know, to me it doesn't make sense.

> How about we go with the functionality and move the naming struggle elsewhere?

 ...

Reply:
I'll dismiss my review.